### PR TITLE
Compatibility with BetterReflection 6.x on ClassFromEnumFactory

### PR DESCRIPTION
**Pull request by samsonasik**
Ref https://github.com/rectorphp/rector-src/actions/runs/3613095940/jobs/6088622166#step:10:22

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
         "symfony/dependency-injection": "6.1.*",
         "symfony/finder": "^6.2",
         "symfony/string": "^6.2",
-        "symplify/autowire-array-parameter": "^11.1.17",
+        "symplify/autowire-array-parameter": "dev-main as 11.1.17",
```
Comment (samsonasik) [inline composer.json]: @TomasVotruba this is needed while wait for new release of symplify/autowire-array-parameter so the build will working

**Comment by TomasVotruba** [comment]
Tests for array-autowire-parameter should be added here too, so we know when it breaks in the future

**Comment by samsonasik** [comment]
I will try move the test as well 👍 Let's try make symplify builds green first

**Comment by samsonasik** [comment]
https://github.com/rectorphp/rector-src/pull/3152

**Pull request merged**